ext/socket/lib/socket.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 require 'socket.so'
-require 'io/wait'
+require 'io/wait' unless ::IO.public_instance_methods(false).include? :wait_readable
```

Comment:
Why not `::IO.method_defined?(:wait_readable, false)`?

Reply:
Okay, my bad, let me fix it.

Reply:
Oh, looks like @nobu already did it.

However, what about all of these?

```
lib/resolv.rb:
  4  require 'timeout'
  5: require 'io/wait'
  6  

lib/drb/drb.rb:
  49  require 'socket'
  50: require 'io/wait'
  51  require 'monitor'

lib/net/protocol.rb:
  23  require 'timeout'
  24: require 'io/wait'
  25  

lib/reline/ansi.rb:
  1  require 'io/console'
  2: require 'io/wait'
  3  require 'timeout'

lib/reline/general_io.rb:
  1  require 'timeout'
  2: require 'io/wait'
  3  

spec/ruby/library/io-wait/wait_readable_spec.rb:
  2  
  3: require 'io/wait'
  4  

spec/ruby/library/io-wait/wait_writable_spec.rb:
  2  
  3: require 'io/wait'
  4  

test/-ext-/thread_fd/test_thread_fd_close.rb:
  3  require '-test-/thread_fd'
  4: require 'io/wait'
  5  

test/io/wait/test_io_wait_uncommon.rb:
  2  require 'test/unit'
  3: require 'io/wait'
  4  

test/io/wait/test_io_wait.rb:
  6  begin
  7:   require 'io/wait'
  8  rescue LoadError

test/io/wait/test_ractor.rb:
  3  require 'rbconfig'
  4: require 'io/wait'
  5  

test/ruby/test_process.rb:
  5  require 'timeout'
  6: require 'io/wait'
  7  require 'rbconfig'

tool/lib/webrick/httpserver.rb:
  11  
  12: require 'io/wait'
  13  require_relative 'server'
```

Reply:
I guess my question is, do we want to remove these? For internal test code, I can see removal being acceptable. For external library code, it might require backwards compatibility in the same way.

Reply:
> For external library code, it might require backwards compatibility in the same way.

Agreed.

Re test code, maybe specs should also be backwards compatible?  Spec code should probably test both with and without io/wait loaded?  Never checked, does io/wait overwrite the methods in current IO?